queries.sql

```diff
@@ -15,3 +15,55 @@ SELECT * from animals WHERE neutered = TRUE;
 SELECT * from animals WHERE name != 'Gabumon';
 
 SELECT * from animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+--1 Transaction 
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+--2 transaction
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals; 
+
+--3 Transaction 
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+-- 4 Transaction 
+BEGIN TRANSACTION;
+DELETE from animals WHERE date_of_birth > '2022/01/01';
+SAVEPOINT animals_after_2022;
+UPDATE animals SET weight_kg = (weight_kg * -1);
+ROLLBACK TO SAVEPOINT animals_after_2022;
+UPDATE animals SET weight_kg = (weight_kg * -1) WHERE weight_kg < 0;
+COMMIT;
+SELECT * from animals;
+
+--How many animals are there?
+SELECT COUNT(*) from animals;
+
+--How many animals have never tried to escape?
+SELECT COUNT(*) from animals WHERE escape_attempts = 0;
+
+--What is the average weight of animals?
+SELECT AVG(weight_kg) from animals;
+
+--Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts) from animals GROUP BY neutered;
+
+--What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg), MAX(weight_kg) from animals GROUP BY species;
+
+--What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) from animals 
+WHERE date_of_birth BETWEEN '1990/01/01' AND '2000/01/01'
```

Comment:
- [ ] Here we have to find bw 1990 to 2000. so kindly use 
`WHERE date_of_birth BETWEEN '1990/01/01' AND '2000/12/31'`